graphql/resolvers.go

```diff
@@ -2530,6 +2524,138 @@ func (r *queryResolver) BbGetCreatedTickets(ctx context.Context, taskID string)
 	return createdTickets, nil
 }
 
+// Will return an array of activated and unactivated versions
+func (r *queryResolver) MainlineCommits(ctx context.Context, options MainlineCommitsOptions) (*MainlineCommits, error) {
+
+	opts := model.MainlineCommitVersionOptions{
+		Activated:       true,
+		SkipOrderNumber: 0,
+		Limit:           0,
+	}
+	if options.Limit != nil {
+		opts.Limit = *options.Limit
+	}
+	if options.SkipOrderNumber != nil {
+		opts.SkipOrderNumber = *options.SkipOrderNumber
+	}
+
+	activatedVersions, err := model.GetMainlineCommitVersionsWithOptions(options.ProjectID, opts)
+	if err != nil {
+		return nil, ResourceNotFound.Send(ctx, fmt.Sprintf("Error getting activated versions %s", err.Error()))
+	}
+	opts = model.MainlineCommitVersionOptions{
+		Activated:       false,
+		SkipOrderNumber: 0,
+		Limit:           0,
+	}
+	if options.Limit != nil {
+		opts.Limit = *options.Limit
+	}
+	if options.SkipOrderNumber != nil {
+		opts.SkipOrderNumber = *options.SkipOrderNumber
+	}
+
+	unactivatedVersions, err := model.GetMainlineCommitVersionsWithOptions(options.ProjectID, opts)
+	if err != nil {
+		return nil, ResourceNotFound.Send(ctx, fmt.Sprintf("Error getting unactivated versions %s", err.Error()))
+	}
+	versions := append(activatedVersions, unactivatedVersions...)
+	sort.Slice(versions, func(i, j int) bool {
+		return versions[i].RevisionOrderNumber > versions[j].RevisionOrderNumber
+	})
+	var mainlineCommits MainlineCommits
+	activatedVersionCount := 0
+	for _, v := range versions {
+		if activatedVersionCount == *options.Limit {
```

Comment:
preference for FromIntPointer since it's safer
Do we set options.Limit to the default anywhere here or is this potentially just 0?

Reply:
This hasn't been answered/updated ^

Reply:
You are right i assumed gqlgen would have automatically applied a default for this field[ based off of us supplying a default value in the schema definition](https://github.com/evergreen-ci/evergreen/pull/4663/files#diff-5424116e8713d835fc94a4a0b222638eebf7c61231381c46bee66d6e549855f4R176). But that doesn't appear to be the case.